astropy/units/physical.py

```diff
@@ -527,4 +528,32 @@ def get_physical_type(obj):
 for unit, physical_type in _units_and_physical_types:
     def_physical_type(unit, physical_type)
 
+
+# This generates a docstring addition for this module that describes all of the
+# standard physical types defined here.
+if __doc__ is not None:
+    docstring = io.StringIO()
+
+    docstring.write("""
+
+.. list-table:: Defined Physical Types
+   :header-rows: 1
+   :widths: 30 10 50
+
+   * - Physical type
+     - Unit
+     - Other physical type(s) with same unit
+""")
+
+    for ptype_name in sorted(_name_physical_mapping.keys()):
+        physical_type = _name_physical_mapping[ptype_name]
+        docstring.write(f"""
+   * - {ptype_name}
+     - :math:`{physical_type._unit.to_string('latex')[1:-1]}`
+     - {', '.join([name for name in physical_type if name != ptype_name])}
+""")
```

Comment:
Bikeshedding warning 😉 

Which is less ugly here? For some reason the """) at the start of the line is jarring to me, but have to admit that my suggestion might be equally abhorrent 😅. If you go with mine here, change the block above to be similar?
```suggestion
        docstring.write("\n".join([
            f"   * - {ptype_name}",
            f"     - :math:`{physical_type._unit.to_string('latex')[1:-1]}`",
            f"     - {', '.join([name for name in physical_type if name != ptype_name])}"
        ])
```